lib/game.rb

```diff
@@ -38,10 +54,27 @@ def switch_players
   def ask_move(player)
     output("#{player.name}'s move", true)
     output('Enter your next move (1-9): ')
-    move = input
+    move = input.to_i
+    until valid_move?(move)
+      output('Invalid move. Try again: ')
+      move = input.to_i
+    end
+    move
   end
 
-  def won?; end
+  def valid_move?(move)
+    move > 0 && move < 10 && !@current_player.positions.include?(move) && !@other_player.positions.include?(move)
+  end
 
-  def full?; end
-end
\ No newline at end of file
+  def won_by
+    winning_sets = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]]
```

Comment:
Probably winning_sets should be included as an attribute of the Game class rather than be declared everytime you call this `won_by` method.

Reply:
Moved it to a class constant.